d2/src/main/java/com/linkedin/d2/xds/balancer/DualReadZkAndXdsLoadBalancerFactory.java

```diff
@@ -33,17 +34,20 @@ public class DualReadZkAndXdsLoadBalancerFactory implements LoadBalancerWithFaci
   private final LoadBalancerWithFacilitiesFactory _zkLbFactory;
   private final LoadBalancerWithFacilitiesFactory _xdsLbFactory;
   private final DualReadStateManager _dualReadStateManager;
+  private final NewLoadBalancerTaskthreadPool _newNewLoadBalancerTaskThreadPool;
 
   public DualReadZkAndXdsLoadBalancerFactory(@Nonnull DualReadStateManager dualReadStateManager)
```

Comment:
See my comment above, you will have to inject a threadpool into this constructor. The problem is that it's backwards incompatible to add a constructor parameter, so you will have to do it like this:
```java
  @Deprecated
  public DualReadZkAndXdsLoadBalancerFactory(@Nonnull DualReadStateManager dualReadStateManager) {
    this(dualReadStateManager, Executors.newSingleThreadExecutor());
  }

  public DualReadZkAndXdsLoadBalancerFactory(@Nonnull DualReadStateManager dualReadStateManager, ExecutorService executor) {
    ...
```

Reply:
Thanks. That makes sense. For backward compatible, what I have done:
- use `D2ClientConfig` , which could be wired with `container`
- If there is no `ThreadPool` setted by `container`, the `DualReadLoadBalancer` will create a default `ThreadPool`